### ISSUE TITLE
Installing SciPy using nagfor: Option warning: -c option specified twice, continuing

### ISSUE THREAD

**Issue by Brunochris13** [numpy/numpy#17951]
### Reproducing code example:

The warnings happens when I try to install the SciPy library from source, using the NAG's Fortran Compiler(nagfor). I installed numpy using pip and cloned the repository of SciPy. I tried to isntall it using this command
```
python setup.py config_fc --fcompiler=nagfor install
```
I then located the error to numpy/distutils/fcompiler/\_\_init\_\_.py in the _compile method. 
```
        command = compiler + cc_args + extra_flags + s_args + o_args \
                  + extra_postargs + extra_compile_args
```
Both **cc_args** and **s_args** contain  **'-c'** flag, which causes this warnings.

One possible solution was to remove it from s_args but that might cause problems for other compilers. Another solution might be to remove it from cc_args, but I do not know where it specified. Last of my ideas is to have an if and specifically remove the '-c' flag for the nagfor, but this doesn't seem like the most optimal solution.

**Note:** In numpy/distutils/fcompiler/nag.py at line 22 I changed the line from
```
        return ['']
```
to
```
        return []
```
to remove 1 of the 2 errors (It made the nag compiler try to compile an empty file). I have created a Pull Request for this solution.

### Error message:

```
...
Option warning: -c option specified twice, continuing
...
```

### NumPy/Python version information:

1.19.4 3.9.0 (default, Oct  7 2020, 07:39:02) 
[GCC 6.3.1 20161221 (Red Hat 6.3.1-1)]

**Comment by dimpase** [comment]
it looks like a kind of bug to me that scipy makes no effort to use a standard way to supply Fortran, via `FC` env var.
It's perfectly possible to buld numpy with `nagfor` like this:
```
FC=nagfor python3 setup.py build -j 4 install --prefix $HOME/.local
```
But scipy just ingores `FC`, and looks for `gfortran`.

**Comment by dimpase** [comment]
I do get errors in `integrate/odepack/srcma.f` (the log is a bit messy, as it builds two files in parallel, but still
```
Error: scipy/integrate/odepack/srcma.f, line 37: First subscript (50) is greater than upper bound (1) for array ISAV
```

a bit more context:

```
...
nagfor:f77: scipy/integrate/odepack/srcma.f
nagfor:f77: scipy/integrate/odepack/stoda.f
Option warning: -c option specified twice, continuing
NAG Fortran Compiler Release 7.0(Yurakucho) Build 7036
Evaluation trial version of NAG Fortran Compiler Release 7.0(Yurakucho) Build 7036
Option warning: -c option specified twice, continuingObsolescent: scipy/integrate/odepack/srcma.f, line 1: Fixed source form

NAG Fortran Compiler Release 7.0(Yurakucho) Build 7036
Evaluation trial version of NAG Fortran Compiler Release 7.0(Yurakucho) Build 7036
Obsolescent: scipy/integrate/odepack/stoda.f, line 1: Fixed source form
Obsolescent: scipy/integrate/odepack/srcma.f, line 27: DO 10 ends neither with CONTINUE nor ENDDO
Obsolescent: scipy/integrate/odepack/srcma.f, line 29: DO 15 ends neither with CONTINUE nor ENDDO
Obsolescent: scipy/integrate/odepack/srcma.f, line 32: DO 20 ends neither with CONTINUE nor ENDDO
Obsolescent: scipy/integrate/odepack/srcma.f, line 34: DO 25 ends neither with CONTINUE nor ENDDO
EOrrbosro:l escescipy/integrate/odepack/srcma.f, line 36: ntF:i rsscipy/integrate/odepack/stoda.f, line 155: t subDsOc rip10t  en(ds49 )n eiist hgerre awtietrh  tChOaNnT IuNpUpEe rn obro uENDDO
Obsolescent: scipy/integrate/odepack/stoda.f, line 158: DO 20 ends neither with CONTINUE nor ENDDO
Obsolescent: scipy/integrate/odepack/stoda.f, line 183: DO 155 ends neither with CONTINUE nor ENDDO
Obsolescent: scipy/integrate/odepack/stoda.f, line 188: Computed GOTO statement
nd (1) for array ISAV
Error: scipy/integrate/odepack/srcma.f, line 37: First subscript (50) is greater than upper bound (1) for array ISAV
ObsOoblseoslceesncten:t : scipy/integrate/odepack/stoda.f, line 219: scipy/integrate/odepack/srcma.f, line 42: DDOO  180110  eennddss  nneeiitthheerr  wwiitthh  CCOONNTTIINNUUEE  nnoorr  EENNDDDDOO

Obsolescent: scipy/integrate/odepack/stoda.f, line 219: 180 is aO bsshoalreeds cDeOn tt:e rscipy/integrate/odepack/srcma.f, line 44: DO 115 emnds neiintahteiro nw iltahb eClO
NTINUE nor ENDDO
Obsolescent: scipy/integrate/odepack/srcma.f, line 47: DO O120b seonldess cneenitt:h erscipy/integrate/odepack/stoda.f, line 240:  DwOi t210h  eCnOdNsT InNeUiEt hneorr  wEiNtDhD OC
ONTINUE nor OEbNsDoDlOe
scent: scipy/integrate/odepack/srcma.f, line 49: DO 125 ends neithOebrs owlietshc eCnOtN:T Iscipy/integrate/odepack/stoda.f, line 253: NDO 230 ends neither with CONTINUE nor ENUDDOE
 nor ENDDO
Error: scipy/integrate/odepack/srcma.f, line 51: First subscript (49) is greater than upper bounOdb s(o1l)e for array ISAV
Error: scipy/integrate/odepack/srcma.f, line 52: First subscript (50) is greater than upper bound (1) for array ISAV
[NAG Fortran Compiler error scent: scipy/integrate/odepack/stoda.f, line 275: termination, 4 eDOrrors, 9 warnings]
 260 ends neither with CONTINUE nor ENDDO
Obsolescent: scipy/integrate/odepack/stoda.f, line 283: DO 290 ends neither with CONTINUE nor ENDDO
Obsolescent: scipy/integrate/odepack/stoda.f, line 287: DO 300 ends neither with CONTINUE nor ENDDO
Obsolescent: scipy/integrate/odepack/stoda.f, line 295: DO 360 ends neither with CONTINUE nor ENDDO
...
```

**Comment by dimpase** [comment]
a better log:
```
nagfor:f77: scipy/integrate/odepack/srcma.f
Option warning: -c option specified twice, continuing
NAG Fortran Compiler Release 7.0(Yurakucho) Build 7036
Evaluation trial version of NAG Fortran Compiler Release 7.0(Yurakucho) Build 7036
Obsolescent: scipy/integrate/odepack/srcma.f, line 1: Fixed source form
Obsolescent: scipy/integrate/odepack/srcma.f, line 27: DO 10 ends neither with CONTINUE nor ENDDO
Obsolescent: scipy/integrate/odepack/srcma.f, line 29: DO 15 ends neither with CONTINUE nor ENDDO
Obsolescent: scipy/integrate/odepack/srcma.f, line 32: DO 20 ends neither with CONTINUE nor ENDDO
Obsolescent: scipy/integrate/odepack/srcma.f, line 34: DO 25 ends neither with CONTINUE nor ENDDO
Error: scipy/integrate/odepack/srcma.f, line 36: First subscript (49) is greater than upper bound (1) for array ISAV
Error: scipy/integrate/odepack/srcma.f, line 37: First subscript (50) is greater than upper bound (1) for array ISAV
Obsolescent: scipy/integrate/odepack/srcma.f, line 42: DO 110 ends neither with CONTINUE nor ENDDO
Obsolescent: scipy/integrate/odepack/srcma.f, line 44: DO 115 ends neither with CONTINUE nor ENDDO
Obsolescent: scipy/integrate/odepack/srcma.f, line 47: DO 120 ends neither with CONTINUE nor ENDDO
Obsolescent: scipy/integrate/odepack/srcma.f, line 49: DO 125 ends neither with CONTINUE nor ENDDO
Error: scipy/integrate/odepack/srcma.f, line 51: First subscript (49) is greater than upper bound (1) for array ISAV
Error: scipy/integrate/odepack/srcma.f, line 52: First subscript (50) is greater than upper bound (1) for array ISAV
[NAG Fortran Compiler error termination, 4 errors, 9 warnings]
```

(this is all with the latest numpy and latest scipy, on Debian 10 x86_64)

**Comment by dimpase** [comment]
I do not know why this pretty standard Fortran 77 way to pass arrays to a subroutine leads to these errors there, must be a "feature" of `nagfor`. How does one specify passing an array in an .f-file (i.e. Fortran 77)?

**Comment by dimpase** [comment]
this particular issue can be patched as follows
```diff
diff --git a/scipy/integrate/odepack/srcma.f b/scipy/integrate/odepack/srcma.f
index 62c07a571..aa3ba1e22 100644
--- a/scipy/integrate/odepack/srcma.f
+++ b/scipy/integrate/odepack/srcma.f
@@ -11,12 +11,11 @@ c        job  = 1 if common is to be saved (written to rsav/isav)
 c        job  = 2 if common is to be restored (read from rsav/isav)
 c        a call with job = 2 presumes a prior call with job = 1.
 c-----------------------------------------------------------------------
-      integer isav, job
+      integer isav(*), job
       integer ieh, ils, ilsa
       integer i, lenrls, lenils, lenrla, lenila
-      double precision rsav
+      double precision rsav(*)
       double precision rls, rlsa
-      dimension rsav(1), isav(1)
       common /ls0001/ rls(218), ils(39)
       common /lsa001/ rlsa(22), ilsa(9)
       common /eh0001/ ieh(2)
```

the next error:
```
nagfor:f77: scipy/sparse/linalg/eigen/arpack/ARPACK/SRC/cnaitr.f
Option warning: -c option specified twice, continuing
NAG Fortran Compiler Release 7.0(Yurakucho) Build 7036
Evaluation trial version of NAG Fortran Compiler Release 7.0(Yurakucho) Build 7036
Warning: scipy/sparse/linalg/eigen/arpack/ARPACK/SRC/cnaitr.f, line 850: Unused dummy variable NB
Obsolescent: scipy/sparse/linalg/eigen/arpack/ARPACK/SRC/cnaitr.f, line 208: Fixed source form
Error: scipy/sparse/linalg/eigen/arpack/ARPACK/SRC/cnaitr.f, line 666: Inconsistent structure for arg 3 in call to SVOUT
Error: scipy/sparse/linalg/eigen/arpack/ARPACK/SRC/cnaitr.f, line 737: Inconsistent structure for arg 3 in call to SVOUT
[NAG Fortran Compiler error termination, 2 errors, 2 warnings]
error: Command "/usr/local/bin/nagfor -fixed -O4 -Iscipy/sparse/linalg/eigen/arpack/ARPACK/SRC -I/home/dimpase/.local/lib/python3.7/site-packages/numpy-1.21.0.dev0+137.g91d9bbebc-py3.7-linux-x86_64.egg/numpy/core/include -c -c scipy/sparse/linalg/eigen/arpack/ARPACK/SRC/cnaitr.f -o build/temp.linux-x86_64-3.7/scipy/sparse/linalg/eigen/arpack/ARPACK/SRC/cnaitr.o" failed with exit status 2
```

**Comment by mattip** [comment]
Is there a reason this issue is here and not on the scipy/scipy issue tracker?

**Comment by dimpase** [comment]
> Is there a reason this issue is here and not on the scipy/scipy issue tracker?

the root of the original problem (inability to use natfor for building scipy) was in numpy, and fixed by #17950.
I don't know why @Brunochris13 opened this issue here, and I didn't pay attention to it, sorry.

I agree this ought to be migrated to scipy, as the remaininng problems are almost certainly all in its Fortran code.
Is there an easy way to do it?

**Comment by Brunochris13** [comment]
@matip SciPy uses numPy to build sciPy and from my understanding all the files relating to Fortran and the compiler flags are in numPy.  That is why I created the issue here.

**Comment by dimpase** [comment]
@Brunochris13 - oh, you are right, of course. Let me open another issue on scipy and move all the Fortran language -specific things there. This issue should be dealng with the extra `-c` stuff only.

**Comment by dimpase** [comment]
I've opened https://github.com/scipy/scipy/issues/13222 for nagfor Fortran lang issues, and copied things from here I added without thinking about the proper location.

**Comment by rgommers** [comment]
> it looks like a kind of bug to me that scipy makes no effort to use a standard way to supply Fortran, via `FC` env var.

SciPy now builds with Meson and Meson recognizes `FC`, so I think we can safely close this. Thanks all!
